ovos_dinkum_listener/service.py

```diff
@@ -681,7 +681,35 @@ def _save_stt(self, audio_bytes, stt_meta, save_path=None):
             stt_audio_dir = Path(f"{self.default_save_path}/utterances")
         stt_audio_dir.mkdir(parents=True, exist_ok=True)
 
-        filename = hash_sentence(stt_meta["transcription"])
+        listener = self.config.get("listener")
+
+        # What is a good default?
+        # Where does the documentation for this feature go?
+        default_template = "utterance_{date}_{uuid}"
+        default_template = "{hash}-{uuid}"
+        default_template = "{date:%Y-%m-%dT%H%M%S%z}-{uuid}"
+        default_template = "{hash}-{date:%Y-%m-%dT%H%M%S%z}-{uuid}"
+
+        filename_template = listener.get("filename_template", default_template)
+        formatter = _TemplateFilenameFormatter()
+
+        @formatter.register('hash')
+        def transcription_hash():
+            # Build a hash of the transcription
+            try:
+                # handles legacy API
+                text = stt_meta.get('transcription', [])
+            except KeyError:
+                # handles new API
+                # transcriptions should be : List[Tuple[str, int]]
+                try:
+                    text = stt_meta.get('transcriptions')[0][0]
+                except IndexError:
+                    return 'null'
```

Comment:
do we want to raise the exception here? or just skip saving completely instead of using "null" as name?

Reply:
If the user wants to save the audio, then we should save the audio. This indicates the transcription failed, which I think could be helpful. Especially if you are grouping items based on a prefix

Reply:
fair enough!